packages/ckeditor5-core/src/multicommand.js

```diff
@@ -90,12 +113,15 @@ export default class MultiCommand extends Command {
 	}
 
 	/**
-	 * Returns a first enabled command or undefined if none of them is enabled.
+	 * Returns a first enabled command with highest priority or undefined if none of them is enabled.
 	 *
 	 * @returns {module:core/command~Command|undefined}
 	 * @private
 	 */
 	_getFirstEnabledCommand() {
-		return this._childCommands.find( command => command.isEnabled );
+		const definitonsWithEnabledCommand = this._childCommandsDefinitons.filter( definition => definition.command.isEnabled );
+		const definitionWithHighestPriority = definitonsWithEnabledCommand[ definitonsWithEnabledCommand.length - 1 ];
+
+		return definitionWithHighestPriority && definitionWithHighestPriority.command;
```

Comment:
After changing the order of callbacks on the list, this method would get much simpler:
```suggestion
		return this._childCommandsDefinitons.find( ( { command } ) => command.isEnabled );
```